strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/StoragesConfigurationController.java

```diff
@@ -252,6 +258,13 @@ public ResponseEntity addOrUpdateRepository(@ApiParam(value = "The storageId", r
 
             configurationManagementService.saveRepository(storageId, repository);
 
+            if (repository != null && RepositoryTypeEnum.PROXY.getType().equals(repository.getType()))
+            {
+                proxyRepositoryConnectionPoolConfigurationService.setMaxPerRepository(
+                        repository.getRemoteRepository()
+                                .getUrl(), repository.getHttpConnectionPool().getAllocatedConnections());
```

Comment:
1. `repository.getHttpConnectionPool().getAllocatedConnections()` possible `NullPointerException`
2. We should check if `repository.getHttpConnectionPool().getAllocatedConnections()` is positive. Otherwise, we should not override default max per repository (could you please correct `org.carlspring.strongbox.services.impl.ConfigurationManagementServiceImpl#setProxyRepositoryConnectionPoolConfigurations` implementation as well ?)

Reply:
I added the check for positivity directly in `org.carlspring.strongbox.services.impl.ProxyRepositoryConnectionPoolConfigurationServiceImpl#setMaxPerRepository` because I think it's good to have it in a central place.